### ISSUE TITLE
mrview -load image.nii -threshold_min : exists for overlay, but not for the main image?

### ISSUE THREAD

**Issue by kuz** [MRtrix3/mrtrix3#2485]
I want to visualise quite a few files using command line and capture, and I need to set a threshold on the image, basically
`mrview -load image.nii -threshold_min 0.5 -capture.grab -exit`
but it looks like things like `-overlay.threshold_min`, `-overlay.opacity` only exist for overlay, but cannot be applied to the main image from the command line, is that so or am I missing something?

I also tried
`mrview -load image.nii -imagevisible False -overlay.load image.nii -overlay.threshold_min 0.5 -capture.grab -exit`
but that makes both the image and the overload to disappear

Feel like I am missing something obvious... but was not able to find a way to do this.

**Comment by jdtournier** [comment]
No, you're not missing anything, that functionality doesn't exist for the main image - though it would be trivial to add since the machinery is already in place to support it. However, the simplest way forward for your specific situation is simply to apply the threshold to the image _before_ feeding it though to mrview, using NaN for the background, like so:
```
mrcalc image.nii 0.5 -lt nan image.nii -if - | mrview - -capture.grab -exit
```
or alternatively:
```
mrthreshold image.nii -abs 0.5 -nan - | mrcalc image.nii - -mult - | mrview - -capture.grab -exit
```
There are probably other ways of doing this, but that's the general idea. Does this address your specific use case?

**Comment by kuz** [comment]
Perfect, thanks!
In case someone faces the similar issue, I went with the following workaround:

1. Make an empty `.nii` file with the same header as the image want to overlay
```python
import sys
import numpy as np
import nibabel as nib

img = nib.load('/path/to/the/image.nii')
emptyimg = nib.Nifti1Image(np.zeros(img.shape), img.affine, img.header)
nib.save(emptyimg, '/tmp/empty.nii')
```

2. Load that empty image as main image for MRView
```shell
mrview <...> -load "/tmp/empty.nii" -intensity_range 0.0,0.5 -overlay.load "/path/to/the/image.nii" <...>
```

The solution by @jdtournier above might be more straightforward though
